### ISSUE TITLE
Constant conflicts after update

### ISSUE THREAD

**Issue by johnnyfreeman** [joeferraro/MavensMate-Atom#148]
I upgraded to `0.0.16` this morning and have gotten constant file conflicts since then. 

> A conflict has been detected between the local and server copy. Please refresh the metadata from server to continue.

Is it possible for me to rollback to version `0.0.15`?

**Comment by joeferraro** [comment]
uncheck `mm_compile_check_conflicts` in the package settings

**Comment by joeferraro** [comment]
in order to help debug this, can you provide more information?

**Comment by johnnyfreeman** [comment]
@joeferraro I sure can!

* MavensMate version: `0.0.16`
* Atom version: `0.179.0`
* OS: OS X Yosemite (10.10.2)

Here is the output from the MM console:

```
[February 16th 2015, 4:21:03 pm]
> Compiling InvoiceController.cls
> A conflict has been detected between the local and server copy. Please refresh the metadata from server to continue.
{
  "result": {
    "checkOnly": false,
    "completedDate": "",
    "createdBy": "",
    "createdByName": "",
    "createdDate": "",
    "details": {
      "componentSuccesses": [],
      "runTestResult": {
        "numFailures": "0",
        "numTestsRun": "0",
        "totalTime": "0.0"
      },
      "componentFailures": [],
      "conflicts": {
        "InvoiceController.cls": {
          "local": {
            "createdById": "005i0000003ZUucAAG",
            "createdByName": "Johnny Freeman",
            "createdDate": "2015-02-11T16:54:50.000Z",
            "fileName": "unpackaged/classes/InvoiceController.cls",
            "fullName": "InvoiceController",
            "id": "01pe00000007X6cAAE",
            "lastModifiedById": "005i0000003ZUucAAG",
            "lastModifiedByName": "Johnny Freeman",
            "lastModifiedDate": "2015-02-16T20:48:30.000Z",
            "manageableState": "unmanaged",
            "type": "ApexClass",
            "mmState": "clean"
          },
          "remote": {
            "attributes": {
              "type": "ApexClass",
              "url": "/services/data/v32.0/sobjects/ApexClass/01pe00000007X6cAAE"
            },
            "LastModifiedById": "005i0000003ZUucAAG",
            "LastModifiedDate": "2015-02-16T21:20:52.000+0000",
            "LastModifiedBy": {
              "attributes": {
                "type": "User",
                "url": "/services/data/v32.0/sobjects/User/005i0000003ZUucAAG"
              },
              "Name": "Johnny Freeman"
            },
            "Body": "public class InvoiceController {\n\n\tpublic invoiceit_s__Invoice__c invoice { get; set; }\n\tpublic List<invoiceit_s__Invoice_Lines__c> invoiceLines { get; set; }\n\tpublic Invoice_Dynamic_Visual_Content__c dynamicVisualContent { get; set; }\n\tpublic SessionDataController session { get; set; }\n\n\tpublic InvoiceController () {\n\n\t\t// authenticate session\n\t\tthis.session = new SessionDataController();\n\t\tif (!this.session.validated) this.redirectToSSOHandler();\n\n\t\t// Query for the invoice (and the invoice line items in subquery)\n\t\tthis.invoice = [\n\t\t\tSELECT Id,\n\t\t\t\tName,\n\t\t\t\tinvoiceit_s__Account__r.Cert_Number__c,\n\t\t\t\tinvoiceit_s__Account__r.ACCU_Account_Number__c,\n\t\t\t\tinvoiceit_s__Billing_Contact__r.Email,\n\t\t\t\tinvoiceit_s__Billing_Contact__r.Full_Name__c,\n\t\t\t\tinvoiceit_s__Billing_Contact__r.Address_Billing__r.Address_Line_1__c,\n\t\t\t\tinvoiceit_s__Billing_Contact__r.Address_Billing__r.Address_Line_2__c,\n\t\t\t\tinvoiceit_s__Billing_Contact__r.Address_Billing__r.City__c,\n\t\t\t\tinvoiceit_s__Billing_Contact__r.Address_Billing__r.State__c,\n\t\t\t\tinvoiceit_s__Billing_Contact__r.Address_Billing__r.Zip__c,\n\t\t\t\tinvoiceit_s__Account__c,\n\t\t\t\tinvoiceit_s__Amount__c,\n\t\t\t\tinvoiceit_s__Billing_Contact__c,\n\t\t\t\tinvoiceit_s__Cancellation_Reason__c,\n\t\t\t\tinvoiceit_s__CurrencyL__c,\n\t\t\t\tinvoiceit_s__Custom_Invoice_Number__c,\n\t\t\t\tinvoiceit_s__Delivery_Type__c,\n\t\t\t\tinvoiceit_s__Discount_Value__c,\n\t\t\t\tinvoiceit_s__Due_Date__c,\n\t\t\t\tinvoiceit_s__Email_Sent__c,\n\t\t\t\tinvoiceit_s__Exchange_Rate_Applied__c,\n\t\t\t\tinvoiceit_s__Invoice_Currency_Symbol__c,\n\t\t\t\tinvoiceit_s__Invoice_Date__c,\n\t\t\t\tinvoiceit_s__Invoice_Reference__c,\n\t\t\t\tinvoiceit_s__Invoice_Run_Created_By__c,\n\t\t\t\tinvoiceit_s__Invoice_State__c,\n\t\t\t\tinvoiceit_s__Invoice_Status__c,\n\t\t\t\tinvoiceit_s__Net_Total__c,\n\t\t\t\tinvoiceit_s__Number_Of_Invoice_Lines__c,\n\t\t\t\tinvoiceit_s__Order__c,\n\t\t\t\tinvoiceit_s__Paid_Amount__c,\n\t\t\t\tinvoiceit_s__Payment_Request_Amount__c,\n\t\t\t\tinvoiceit_s__Payment_Status__c,\n\t\t\t\tinvoiceit_s__Regular_Invoice__c,\n\t\t\t\tinvoiceit_s__Shipping_Contact__c,\n\t\t\t\tinvoiceit_s__Status_Flag__c,\n\t\t\t\tinvoiceit_s__Target_Date__c,\n\t\t\t\tinvoiceit_s__Tax__c,\n\t\t\t\tinvoiceit_s__Total_Invoice_Value__c,\n\t\t\t\tinvoiceit_s__Unpaid_Amount__c,\n\t\t\t\tinvoiceit_s__Unpost_Reason__c,\n\t\t\t\tinvoiceit_s__VAT__c,\n\t\t\t\t(\n\t\t\t\t\tSELECT Id,\n\t\t\t\t\t\tinvoiceit_s__Accounting_Code__c,\n\t\t\t\t\t\tinvoiceit_s__Calculated_Quantity__c,\n\t\t\t\t\t\tinvoiceit_s__Charge_Date__c,\n\t\t\t\t\t\tinvoiceit_s__CurrencyL__c,\n\t\t\t\t\t\tinvoiceit_s__Discount_Value__c,\n\t\t\t\t\t\tinvoiceit_s__Grouped_Quantity__c,\n\t\t\t\t\t\tinvoiceit_s__Invoice__c,\n\t\t\t\t\t\tinvoiceit_s__Job_Rate_Plan_Charge__c,\n\t\t\t\t\t\tinvoiceit_s__Job__c,\n\t\t\t\t\t\tinvoiceit_s__Line_Amount__c,\n\t\t\t\t\t\tinvoiceit_s__Line_Name__c,\n\t\t\t\t\t\tinvoiceit_s__Net_Total__c,\n\t\t\t\t\t\tinvoiceit_s__Product_Name__c,\n\t\t\t\t\t\tinvoiceit_s__Product_Rate_Plan_Charge__c,\n\t\t\t\t\t\tinvoiceit_s__Product__c,\n\t\t\t\t\t\tinvoiceit_s__Quantity__c,\n\t\t\t\t\t\tinvoiceit_s__Sequence_No__c,\n\t\t\t\t\t\tinvoiceit_s__Service_End_Date__c,\n\t\t\t\t\t\tinvoiceit_s__Service_Start_Date__c,\n\t\t\t\t\t\tinvoiceit_s__Status__c,\n\t\t\t\t\t\tinvoiceit_s__Tax_Code__c,\n\t\t\t\t\t\tinvoiceit_s__Tax__c,\n\t\t\t\t\t\tinvoiceit_s__Total_Line_Amount__c,\n\t\t\t\t\t\tinvoiceit_s__Unit_of_Measurement__c,\n\t\t\t\t\t\tinvoiceit_s__Unit_Price__c,\n\t\t\t\t\t\tinvoiceit_s__VAT__c,\n\t\t\t\t\t\tName\n\t\t\t\t\tFROM invoiceit_s__Invoice_Lines__r\n\t\t\t\t\tWHERE invoiceit_s__Invoice__c = :ApexPages.currentPage().getParameters().get('id')\n\t\t\t\t\tORDER BY invoiceit_s__Sequence_No__c ASC NULLS FIRST\n\t\t\t\t)\n\t\t\tFROM invoiceit_s__Invoice__c\n\t\t\tWHERE Id = :ApexPages.currentPage().getParameters().get('id')\n\t\t];\n\n\t\tthis.invoiceLines = new List<invoiceit_s__Invoice_Lines__c>();\n\n\t\t// if not null\n\t\tif (this.invoice.invoiceit_s__Invoice_Lines__r != null) {\n\t\t\tthis.invoiceLines = this.invoice.invoiceit_s__Invoice_Lines__r;\n\t\t}\n\n\t\t// Query for the monthly dynamic content: left side image, bottom left text about monthly bills metrics, and bottom right Bible verse\n\t\tthis.dynamicVisualContent = [\n\t\t\tSELECT Left_Side_Image_Id__c,\n\t\t\t\tLower_Left_Textbox__c,\n\t\t\t\tLower_Right_Textbox__c,\n\t\t\t\tNumerical_Month__c,\n\t\t\t\tNumerical_Year__c\n\t\t\tFROM Invoice_Dynamic_Visual_Content__c\n\t\t\tWHERE Numerical_Year__c = :String.valueOf(this.invoice.invoiceit_s__Invoice_Date__c.year())\n\t\t\t\tAND Numerical_Month__c = :this.invoice.invoiceit_s__Invoice_Date__c.month()\n\t\t];\n\n\t\t// TODO if this.dynamicVisualContent is null, set the default pictures.\n\t}\n\n\tpublic PageReference redirectToSSOHandler() {\n\t\treturn Page.SSOHandlerPage;\n\t}\n\n\tpublic Boolean isMannaInvoice {\n\t\tget {\n\t\t\tif (this.isMannaInvoice == null) {\n\t\t\t\tthis.isMannaInvoice = true;\n\t\t\t}\n\t\t\treturn this.isMannaInvoice;\n\t\t}\n\t\tset;\n\t}\n\n\tpublic String leftSideImageUrl {\n\t\tget {\n\t\t\treturn '/servlet/servlet.ImageServer?id=' + this.dynamicVisualContent.Left_Side_Image_Id__c + '&oid=' + UserInfo.getOrganizationId();\n\t\t}\n\t\tprivate set;\n\t}\n\n\tpublic List<Manna_Billing_Need__c> mannaBillingNeeds {\n\t\tget {\n\t\t\tif (this.mannaBillingNeeds == null) {\n\t\t\t\tthis.mannaBillingNeeds = [\n\t\t\t\t\tSELECT Id,\n\t\t\t\t\t\tName\n\t\t\t\t\tFROM Manna_Billing_Need__c\n\t\t\t\t\tWHERE CALENDAR_MONTH(Manna_Billing_Run__r.CreatedDate) = :this.invoice.invoiceit_s__Due_Date__c.month()\n\t\t\t\t];\n\t\t\t}\n\t\t\treturn this.mannaBillingNeeds;\n\t\t}\n\t\tprivate set;\n\t}\n}"
          }
        }
      }
    },
    "done": true,
    "id": "",
    "ignoreWarnings": false,
    "lastModifiedDate": "",
    "numberComponentErrors": 0,
    "numberComponentsDeployed": 0,
    "numberComponentsTotal": 0,
    "numberTestErrors": 0,
    "numberTestsCompleted": 0,
    "numberTestsTotal": 0,
    "rollbackOnError": false,
    "runTestsEnabled": "false",
    "startDate": "",
    "status": "Conflict",
    "success": false
  },
  "promiseId": "b61e90f0-b621-11e4-b207-377983d59f5f"
}
```

Is there anything else I can post here that would be useful to you?

**Comment by joeferraro** [comment]
did you refresh the file?

**Comment by johnnyfreeman** [comment]
Yes. I'm sorry I should have explained further. I refresh the file. That action is successful. I hit save. Wait a minute. Hit save again. Conflict error. My only thought is that perhaps the first save is timestamping it incorrectly somehow? I don't know.

**Comment by joeferraro** [comment]
i just pushed a patch that i hopes resolve this issue. if you could, update to 0.0.17, ensure `mm_compile_check_conflicts` is set to true, and let me know whether it's resolved. 

if it is not resolved, simply uncheck `mm_compile_check_conflicts` for the time-being until we can do a deeper dive

**Comment by johnnyfreeman** [comment]
Excellent! Thank you. I'll let you know tomorrow if it solves the issue for me.

**Comment by johnnyfreeman** [comment]
Went ahead and upgraded and did a quick test. Here's what I did step by step:

* Upgraded MM package
* Enabled conflict setting
* Hit save in `InvoiceController.cls` (got conflict error here)
* Refreshed from server (console shows refresh action executed three times)
* Hit save again in `InvoiceController.cls` (got another conflict error here)

Here's the console output for all steps above:

```
[February 16th 2015, 5:33:46 pm]
> Compiling InvoiceController.cls
> A conflict has been detected between the local and server copy. Please refresh the metadata from server to continue.

[February 16th 2015, 5:33:34 pm]
> Compiling InvoiceController.cls
> Compile Completed Successfully

[February 16th 2015, 5:33:22 pm]
> Refreshing InvoiceController.cls
> Metadata successfully refreshed

[February 16th 2015, 5:33:22 pm]
> Refreshing InvoiceController.cls
> Metadata successfully refreshed

[February 16th 2015, 5:33:22 pm]
> Refreshing InvoiceController.cls
> Metadata successfully refreshed

[February 16th 2015, 5:33:12 pm]
> Compiling InvoiceController.cls
> A conflict has been detected between the local and server copy. Please refresh the metadata from server to continue.

MavensMate project initialized successfully. Happy coding!

Initializing MavensMate project...
```

**Comment by nem604** [comment]
I'm also experiencing the same behaviour as @johnnyfreeman 
MavensMate version: 0.0.17
Atom version: 0.179.0
OS: OS X Yosemite (10.10.2)

**Comment by nem604** [comment]
@joeferraro It appears that version 0.018 resolved this behaviour.

**Comment by joeferraro** [comment]
0.0.18 is a temporary fix that disables conflict checking by default until we can delve into this ticket. please provide as much information as possible so we can diagnose and patch properly.

**Comment by mh2292** [comment]
If I compile the same file twice in a row, I receive the conflict error after the second compile, even when making no changes. If I refresh the file immediately after each compile, the issue seems to be resolved.

**Comment by joeferraro** [comment]
this should be resolved in v1.0.0. please feel free to re-open if you're still experiencing this issue. thanks!
